libraries/AP_Networking/AP_Networking_SpeedTest.cpp

```diff
@@ -0,0 +1,216 @@
+    // @Range: 0 255
+    // @User: Advanced
+    AP_GROUPINFO("DST_ADDR1", 3,  AP_Networking_SpeedTest,    _params.ip[1],   255),
+
+    // @Param: DST_ADDR2
+    // @DisplayName: IPv4 Dest Addr 3rd byte
+    // @Description: IPv4 Destination Address 3rd byte. Example: xxx.xxx.13.xxx. Default IP is Broadcast: 255.255.255.255
+    // @Range: 0 255
+    // @User: Advanced
+    AP_GROUPINFO("DST_ADDR2", 4,  AP_Networking_SpeedTest,    _params.ip[2],   255),
+
+    // @Param: DST_ADDR3
+    // @DisplayName: IPv4 Dest Addr LSB
+    // @Description: IPv4 Destination Address LSB. Example: xxx.xxx.xxx.123. Default IP is Broadcast: 255.255.255.255
+    // @Range: 0 255
+    // @User: Advanced
+    AP_GROUPINFO("DST_ADDR3", 5,  AP_Networking_SpeedTest,    _params.ip[3],   255),
+
+    // @Param: SIZE
+    // @DisplayName: SpeedTest payload size
+    // @Description: SpeedTest payload size per packet. The largest UDP payload capacity for Ethernet is 1472. That's 1500 minus UDP header. Going above this will cause packet fragmentation. Jumbo frames are not supported.
+    // @Range: 0 65535
+    // @User: Advanced
+    AP_GROUPINFO("SIZE", 7, AP_Networking_SpeedTest, _params.payload_size, AP_NETWORKING_ETHERNET_UDP_PAYLOAD_MAX_SIZE),
+
+    // @Param: CONTENT
+    // @DisplayName: SpeedTest payload content
+    // @Description: SpeedTest payload content. This controls the content of the packet. It doesn't not change any behavior, it's used for debugging if you want to capture and/or view the data in tools such as Wireshark. A value of -1 will fill the packet with the string "ArduPilot Networking SpeedTest". Values 0-255 will populate the packet with that number.
+    // @Range: 0 65535
+    // @User: Advanced
+    AP_GROUPINFO("CONTENT", 8, AP_Networking_SpeedTest, _params.content, -1),
+
+    // @Param: DURATION
+    // @DisplayName: SpeedTest duration
+    // @Description: SpeedTest duration in seconds. Setting the value > 0 will start the speedtest. To protect the network, this value resets to zero on boot and after the Speedtest duration has elapsed.
+    // @Units: s
+    // @User: Standard
+    AP_GROUPINFO("DURATION", 9, AP_Networking_SpeedTest, _params.duration_seconds, 0),
+    
+    AP_GROUPEND
+};
+
+extern const AP_HAL::HAL& hal;
+
+/// Constructor
+AP_Networking_SpeedTest::AP_Networking_SpeedTest(AP_Networking &front, AP_Networking::AP_Networking_State &state,  AP_Networking_Params &params) :
+    AP_Networking_Backend(front, state, params)
+{
+    AP_Param::setup_object_defaults(this, var_info);
+    _state.var_info = var_info;
+}
+
+void AP_Networking_SpeedTest::init()
+{
+    // always set this to 0 so we don't spam on boot
+    _params.duration_seconds.set_and_save(0);
+
+    if (_eth.pcb == nullptr) {
+        _eth.pcb = udp_new_ip_type(IPADDR_TYPE_ANY);
+        if (_eth.pcb != nullptr) {
+            // allow to sending broadcast packets
```

Comment:
Broadcast speedtest sounds like a good way to flood a network. Are you sure this is a good idea?

Reply:
This is why there's a duration thing on it. I'm open to suggestions. And this is likely going to be ripped out in favor of a script